<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>949</width>
    <height>774</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Homography finder</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>951</width>
      <height>731</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Image</string>
       </property>
       <widget class="ControllableLabel" name="imageLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>625</width>
          <height>695</height>
         </rect>
        </property>
        <property name="maximumSize">
         <size>
          <width>625</width>
          <height>695</height>
         </size>
        </property>
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="GridDrawerOpenGLWidget" name="widget" native="true">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>625</width>
          <height>695</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>625</width>
          <height>695</height>
         </size>
        </property>
       </widget>
       <zorder>widget</zorder>
       <zorder>imageLabel</zorder>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="title">
        <string>Control panel</string>
       </property>
       <widget class="QLineEdit" name="loadedImagePathEdit">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>281</width>
          <height>20</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>20</y>
          <width>81</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Loaded image:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_17">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>70</y>
          <width>47</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>degrees</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>70</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Rotation step:</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="angleDegreeSpinBox">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>70</y>
          <width>42</width>
          <height>22</height>
         </rect>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>180</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>2</number>
        </property>
       </widget>
       <widget class="QTabWidget" name="tabWidget">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>200</y>
          <width>281</width>
          <height>471</height>
         </rect>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>1000</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <property name="maximumSize">
          <size>
           <width>300</width>
           <height>16777215</height>
          </size>
         </property>
         <attribute name="title">
          <string>Keyboard control</string>
         </attribute>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>251</width>
            <height>231</height>
           </rect>
          </property>
          <property name="title">
           <string>Rotations</string>
          </property>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>60</y>
             <width>181</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>X counter-clockwise press [a]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_4">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>X axis rotation:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_5">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>40</y>
             <width>141</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>X clockwise press [q]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_7">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>130</y>
             <width>181</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Y counter-clockwise press [d]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_9">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>110</y>
             <width>141</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Y clockwise press [e]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_8">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>90</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Y axis rotation:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_11">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>200</y>
             <width>181</width>
             <height>16</height>
            </rect>
           </property>
           <property name="text">
            <string>Z counter-clockwise press [s]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_12">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>180</y>
             <width>141</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Z clockwise press [w]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_10">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>160</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Z axis rotation:</string>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>250</y>
            <width>251</width>
            <height>131</height>
           </rect>
          </property>
          <property name="title">
           <string>Translation</string>
          </property>
          <widget class="QLabel" name="label_36">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>60</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Translate in X:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_23">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>60</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>positive [u]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_37">
           <property name="geometry">
            <rect>
             <x>19</x>
             <y>20</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Translation step:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_29">
           <property name="geometry">
            <rect>
             <x>190</x>
             <y>20</y>
             <width>47</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>unit(s)</string>
           </property>
          </widget>
          <widget class="QDoubleSpinBox" name="translationStepSpinBox">
           <property name="geometry">
            <rect>
             <x>111</x>
             <y>20</y>
             <width>61</width>
             <height>22</height>
            </rect>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.100000000000000</double>
           </property>
          </widget>
          <widget class="QLabel" name="label_25">
           <property name="geometry">
            <rect>
             <x>160</x>
             <y>60</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>negative [j]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_26">
           <property name="geometry">
            <rect>
             <x>160</x>
             <y>80</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>negative [k]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_38">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>80</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>positive [i]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_39">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>80</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Translate in Y:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_40">
           <property name="geometry">
            <rect>
             <x>160</x>
             <y>100</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>negative [l]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_41">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>100</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>positive [o]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_42">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>100</y>
             <width>71</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Translate in Z:</string>
           </property>
          </widget>
          <widget class="QPushButton" name="getHomographyButton">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>100</y>
             <width>281</width>
             <height>41</height>
            </rect>
           </property>
           <property name="text">
            <string>Get current homography</string>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>390</y>
            <width>251</width>
            <height>51</height>
           </rect>
          </property>
          <property name="title">
           <string>Grid manipulation</string>
          </property>
          <widget class="QLabel" name="label_34">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>41</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Density:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_13">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>20</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>more dense [m]</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_14">
           <property name="geometry">
            <rect>
             <x>150</x>
             <y>20</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>less dense [n]</string>
           </property>
          </widget>
         </widget>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Mouse control</string>
         </attribute>
         <widget class="QGroupBox" name="groupBox_8">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>251</width>
            <height>231</height>
           </rect>
          </property>
          <property name="title">
           <string>Rotations</string>
          </property>
          <widget class="QPushButton" name="pushButton_13">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>40</y>
             <width>75</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>X clockwise</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_14">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>61</y>
             <width>111</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>X counter-clockwise</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_15">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>X axis rotation:</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_15">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>131</y>
             <width>111</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Y counter-clockwise</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_16">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>110</y>
             <width>75</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Y clockwise</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_24">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>90</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Y axis rotation:</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_17">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>201</y>
             <width>111</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Z counter-clockwise</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_18">
           <property name="geometry">
            <rect>
             <x>50</x>
             <y>180</y>
             <width>75</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Z clockwise</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_27">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>160</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Z axis rotation:</string>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox_7">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>240</y>
            <width>251</width>
            <height>131</height>
           </rect>
          </property>
          <property name="title">
           <string>Translation</string>
          </property>
          <widget class="QPushButton" name="pushButton_25">
           <property name="geometry">
            <rect>
             <x>60</x>
             <y>90</y>
             <width>131</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Translate</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_35">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>131</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Type delta in form 'x,y,z': </string>
           </property>
          </widget>
          <widget class="QLineEdit" name="lineEdit_3">
           <property name="geometry">
            <rect>
             <x>20</x>
             <y>50</y>
             <width>211</width>
             <height>20</height>
            </rect>
           </property>
          </widget>
         </widget>
         <widget class="QGroupBox" name="groupBox_9">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>370</y>
            <width>251</width>
            <height>61</height>
           </rect>
          </property>
          <property name="title">
           <string>Grid manipulation</string>
          </property>
          <widget class="QPushButton" name="pushButton_21">
           <property name="geometry">
            <rect>
             <x>160</x>
             <y>20</y>
             <width>75</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>Less dense</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton_22">
           <property name="geometry">
            <rect>
             <x>70</x>
             <y>20</y>
             <width>75</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>More dense</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_31">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>41</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Density:</string>
           </property>
          </widget>
         </widget>
        </widget>
       </widget>
       <widget class="QLabel" name="label_21">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>100</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>pixels</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="gridUnitSizeEdit">
        <property name="geometry">
         <rect>
          <x>100</x>
          <y>100</y>
          <width>61</width>
          <height>20</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_20">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>100</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>Grid unit size:</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>130</y>
          <width>281</width>
          <height>61</height>
         </rect>
        </property>
        <property name="title">
         <string>Perspective settings:</string>
        </property>
        <widget class="QLineEdit" name="perspectiveSettingsEdit">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>161</width>
           <height>20</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QPushButton" name="perspectiveSettingsButton">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>30</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Change</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>949</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen_image"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionOpen_image">
   <property name="text">
    <string>Open image</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ControllableLabel</class>
   <extends>QLabel</extends>
   <header location="global">controllablelabel.h</header>
  </customwidget>
  <customwidget>
   <class>GridDrawerOpenGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">griddraweropenglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>474</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
